pyo3-arrow/src/buffer.rs

```diff
@@ -80,6 +80,10 @@ impl PyArrowBuffer {
         PyBytes::new_bound(py, &self.0)
     }
 
+    fn __len__(&self) -> PyResult<usize> {
+        Ok(self.0.len())
+    }
```

Comment:
This doesn't need to be fallible.
```suggestion
    fn __len__(&self) -> usize {
        self.0.len()
    }
```